api/base/filters.py

```diff
@@ -171,7 +188,11 @@ def param_queryset(self, query_params, default_queryset):
         if fields_dict:
             for field_name, value in fields_dict.items():
                 if self.is_filterable_field(key=field_name):
-                    queryset = queryset.intersection(set(self.get_filtered_queryset(field_name, value, default_queryset)))
+                    filtered_queryset = set(self.get_filtered_queryset(field_name, value, default_queryset))
+                    queryset = [
+                        result for result in queryset
+                        if result in filtered_queryset
+                    ]
```

Comment:
I don't think I'm understanding the value of 192-195. The `default_queryset` is passed in, `queryset = set(default_queryset)`. Then we pass `default queryset` to `self.get_filtered_queryset` and wrap the return in a `set` and assign it to `filtered_queryset`. All well and good, now we have a set of values from `default_queryset` filtered by the requested params. Then we loop through `queryset` which is a set of `default_queryset`, which we passed to `get_filtered_queryset` and see if each value is in the filtered version and reset `queryset` to that combination in a list comprehension. I know it was there before in a different form. I'm just not sure what the value of either of them is.